src/main/java/com/mitchej123/hodgepodge/mixins/early/minecraft/MixinMinecraft_ClearEffectRenderer.java

```diff
@@ -0,0 +1,28 @@
+package com.mitchej123.hodgepodge.mixins.early.minecraft;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.multiplayer.WorldClient;
+import net.minecraft.client.particle.EffectRenderer;
+
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(Minecraft.class)
+public abstract class MixinMinecraft_ClearEffectRenderer {
+
+    @Shadow
+    public EffectRenderer effectRenderer;
+
+    @Inject(
+            method = "loadWorld(Lnet/minecraft/client/multiplayer/WorldClient;Ljava/lang/String;)V",
+            at = @At(value = "HEAD"))
+    private void hodgepodge$fixEffectRendererWorldLeak(WorldClient worldClient, String loadingMessage,
+            CallbackInfo ci) {
+        if (worldClient == null) {
```

Comment:
I see that the vanilla code in this method is using a null check on the effectRenderer field before calling that method, can you do the same thing ?